spring-cloud-dataflow-registry/src/main/java/org/springframework/cloud/dataflow/registry/RdbmsUriRegistry.java

```diff
@@ -65,9 +66,9 @@ public URI find(String name) {
 			uriString = jdbcTemplate.queryForObject(SELECT_URI_SQL, String.class, name);
 		}
 		catch (EmptyResultDataAccessException e) {
-			return null;
+			throw new IllegalArgumentException("No URI is registered for ["+ name + "] "+ e);
 		}
-		return uriString != null ? toUri(uriString) : null;
```

Comment:
I believe `null` or `non-text` values need not be checked here as the `register` operation has validation for those before registering.